### ISSUE TITLE
Switching to application always open new window

### ISSUE THREAD

**Issue by chiragparekh** [Wox-launcher/Wox#1922]
Hello,

I am using wox but I am facing one issue. Switching to one application to another it always open  new window.

For example if I open Chrome using wox and again if I search chrome in wox instead of focusing it to already opened window it opens new window. Is there any solution for it?

**Comment by chris-peng** [comment]
You can try this plugin.[Switcheroo](http://www.wox.one/plugin/52)

**Comment by nevara** [comment]
I'm trying to use asterisk to switch to an application without using a action keyword. But that does not work. Any pointers?

**Comment by vladdeSV** [comment]
As an Alfred user, switching between applications is essential for my workflow. Currently I do not see Wox as an alternative because of this missing feature.

Using the Switcheroo plugin is not an alternative *for me* because of the additional command needed to switch.

With Alfred, to open Spotify I did the following:
* Opened Alfred (`alt` + `space`)
* Pressed the S-key (`s`)
* Pressed enter (`enter`)

**Comment by dotCipher** [comment]
Yeah, this is a severe hindrance for productivity and I would also love to see this feature in Wox

**Comment by adilk2005** [comment]
> Hello,
> 
> I am using wox but I am facing one issue. Switching to one application to another it always open new window.
> 
> For example if I open Chrome using wox and again if I search chrome in wox instead of focusing it to already opened window it opens new window. Is there any solution for it?

This video shows you how to do that. https://www.youtube.com/watch?v=55ah-dqIzlI
It's an easy solution. Check it out

**Comment by adilk2005** [comment]
> As an Alfred user, switching between applications is essential for my workflow. Currently I do not see Wox as an alternative because of this missing feature.
> 
> Using the Switcheroo plugin is not an alternative _for me_ because of the additional command needed to switch.
> 
> With Alfred, to open Spotify I did the following:
> 
> * Opened Alfred (`alt` + `space`)
> * Pressed the S-key (`s`)
> * Pressed enter (`enter`)

Yeah, Switcheroo is good. If anyone wants to see it in action, you can see this video. It walks you through it https://www.youtube.com/watch?v=55ah-dqIzlI ... you can jump to 6:42 to see this plugin specifically. Otherwise the video goes into more depth about other features it  has

**Comment by williamgreen** [comment]
Hey, "UX" guy here. It still isn't ideal to have two key combos for this (alt+tab replaced by Switcheroo and alt+space for Wox). I have two machines, one MacOS and one Windows. In an ideal world I could use Switcheroo and Wox on Windows to function in the same way as spotlight or alfred on MacOS - i.e. when performing any normal query on Wox, without the Switheroo prefix 'win', Wox knows to prioritise opening existing windows of that application over opening a new instance of it. To fill the gap this change would leave behind, I'd prefer having to prefix or suffix a query with a command to explicitly start a new instance of an application rather than opening an existing tab. Like "new Spotify" or "new Chrome". I think I'm right in saying that most users would prefer this behaviour.

**Comment by adilk2005** [comment]
> Hey, "UX" guy here. It still isn't ideal to have two key combos for this (alt+tab replaced by Switcheroo and alt+space for Wox). I have two machines, one MacOS and one Windows. In an ideal world I could use Switcheroo and Wox on Windows to function in the same way as spotlight or alfred on MacOS - i.e. when performing any normal query on Wox, without the Switheroo prefix 'win', Wox knows to prioritise opening existing windows of that application over opening a new instance of it. To fill the gap this change would leave behind, I'd prefer having to prefix or suffix a query with a command to explicitly start a new instance of an application rather than opening an existing tab. Like "new Spotify" or "new Chrome". I think I'm right in saying that most users would prefer this behaviour.

Hey, I think this solves that problem. Skip to 6:42. Let me know, if this helps.
[https://www.youtube.com/watch?v=55ah-dqIzlI](https://www.youtube.com/watch?v=55ah-dqIzlI)

**Comment by CyberDoyle** [comment]
Is there still no workaround for this issue??

**Comment by adilk2005** [comment]
> Is there still no workaround for this issue??

@iConFusion22 
This video shows you how to do that. https://www.youtube.com/watch?v=55ah-dqIzlI
It's an easy solution. Check it out

There's a plugin you have to add called Switcheroo... The video will tell you about it.

**Comment by CyberDoyle** [comment]
> This video shows you how to do that. https://www.youtube.com/watch?v=55ah-dqIzlI
> It's an easy solution. Check it out
> 
> There's a plugin you have to add called Switcheroo... The video will tell you about it.

Hi @adilk2005 ,
Thanks for your reply. I have seen the video before. 
The issue I was referring to was that Switcheroo doesn't work across multiple workspaces (Virtual Windows Desktops) and it also isn't very intuitive in comparison to the Alfred 

> switch to an application without using a action keyword.

Using the example from the above thread:

> With Alfred, to open Spotify I did the following:
> 
> * Opened Alfred (`alt` + `space`)
> * Pressed the S-key (`s`)
> * Pressed enter (`enter`)


My issues:
You can't just shortcut into switcheroo and start typing a Program's name and hit enter. Job done. (Excluding the alt+tab overwrite)
You can't switch to a program running in a different workspace.
You must use an "Action keyword" to activate Switcheroo.

TLDR:
Is there a better solution or plugin than Switcheroo?

**Comment by kindadolf** [comment]
I would love to have this feature. It seems to be requested here before https://github.com/Wox-launcher/Wox/issues/1570

**Comment by vladdeSV** [comment]
> or this will be closed in 3 days.

I mean, that is one way to handle suggestions that have not been acknowledged for 6 years 🤔

**Comment by qianlifeng** [comment]
> > or this will be closed in 3 days.
> 
> I mean, that is one way to handle suggestions that have not been acknowledged for 6 years 🤔

There were too many pointless issues before [like this](https://github.com/Wox-launcher/Wox/issues/3921). As in https://github.com/Wox-launcher/Wox/discussions/3937 said, we don't have the resources to deal with v1 bugs now. So we use a github-action bot to deal with issues that go unanswered for too long. If anyone still feels the need for this feature, please comment to avoid a stale

**Comment by vladdeSV** [comment]
That makes total sense. Thank you for the work so far.

I'll also add this is still a feature I wish for 😊

**Comment by qianlifeng** [comment]
Already implemented, try latest build: https://github.com/Wox-launcher/Wox/actions/runs/20636988745 if anyone want to try.

**Comment by stavfx** [comment]
Awesome, I've been waiting a long time for this feature.
However, I've found that it doesn't always bring the app to the foreground, just flashes it red in the taskbar.
The red flashing happens also when it works correctly, which would be nice to get rid of, but that's a minor thing that doesn't actually affect functionality

**Comment by qianlifeng** [comment]
> Awesome, I've been waiting a long time for this feature. However, I've found that it doesn't always bring the app to the foreground, just flashes it red in the taskbar. The red flashing happens also when it works correctly, which would be nice to get rid of, but that's a minor thing that doesn't actually affect functionality

Sorry, I can't reproduce the issue you mentioned at the moment—does it occur under any specific conditions?

**Comment by stavfx** [comment]
I've been away from my windows machine. I'll try to figure out if there's anything more context/logs I can give.

**Comment by qianlifeng** [comment]
Closed for now. If you have any other questions, please open another issue.
